### ISSUE TITLE
Review scope of learning outcomes for Weeks 1-8

### ISSUE THREAD

**Issue by sofer** [foundersandcoders/master-reference#691]
> Parameters for learning outcomes for Weeks 1-8 @NataliaLKB @eliascodes @ZooeyMiller

https://github.com/foundersandcoders/london-programme/issues/477

Others are welcome to self assign.

**Comment by sofer** [comment]
@NataliaLKB @ZooeyMiller I had a quick discussion about this with @eliascodes yesterday.

Broadly speaking, it would probably be sensible to review the scope of the learning outcomes for Weeks 1-8 on a regular basis. This is the first such review.

Here are the learning outcomes for the first 8 weeks:
- [Week 1](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-1/learning-outcomes.md)
- [Week 2](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-2/learning-outcomes.md)
- [Week 3](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-3/learning-outcomes.md)
- [Week 4](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-4/learning-outcomes.md)
- [Week 5](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-5/learning-outcomes.md)
- [Week 6](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-6/learning-outcomes.md)
- [Week 7](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-7/learning-outcomes.md)
- [Week 8](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-8/learning-outcomes.md)

There is also a meta-task of defining exactly how to do a review of learning outcomes. It might look like:
- Assign somebody to facilitate the process
- Open an issue for each of the 8 learning outcomes and assign to each of the reviewers
- Also assign to the relevant week facilitator 
- Agree a deadline for adding comments to each of the issues
- Organise a hangouts after the deadline to discuss each of the 8 open issues
- Agree on any suggested amendments to the learning outcomes
- Assign somebody to action any changes proposed
- Open a pull request for a change to the learning outcomes of the week in question, referencing the original issue
- Assign the PR to the week facilitator, so they can respond and add any relevant material to the week, if needed
- Assign somebody to keep an eye on the open PRs
- Officially end the review process after all the open PRs have been merged

This sounds a bit lengthy, but it might end up being quite short as my guess is that if we do this regularly (once a year?), then there won't be many changes to make.

The point of the exercise is not to make any changes to the week's content, but to spot any obvious gaps in learning outcomes and to add them to the most relevant week, where it might or might not be added to the week's workshop material. In the worst case, students will at least become aware of a topic relevant to their project that they may want to find out about.

At some point, we might want to suggest more radical changes, that would entail a wholesale change to an entire week or the addition of a new one, but we probably won't want to do this too regularly.

Please let me know if you want any assistance with any of this.

**Comment by eliascodes** [comment]
Just putting the relevant section of the minutes here from foundersandcoders/london-programme#423:

> Prepared set of learning outcomes each week: split into 2 parts:
> * Idea that at the end of the week you should have a project-based outcome, a clear goal and problem to solve - 5s mostly, 3 4s and 1 2
>   * You still need a destination on a road trip, clarifying that that destination doesn't have to be very detailed
> * List of specific technologies/tools/topics to be familiar with at the end of the week (clarification that something like 'testing' would be part of this list) -
>   * List of what you want to get out of the journey (note that at this stage there is a range here around how specific or broad it is)
>   * Points noted from @NataliaLKB and @adamdiy from FAC3 around how useful it is to let people choose their own technology for their learning and for the lack of stipulation of this 'in the real world' - point around how this affects the consistency of Founders & Coders graduates and the dilution
>   * We currently don't share a common understanding of what this sentence means and therefore can't fist to five it at the moment; this is a key discussion and will need to be picked up separately

**Comment by rebeccaradding** [comment]
FYI this is still open: #48

**Comment by eliascodes** [comment]
I think maybe before discussing learning outcomes for each week individually, we can have a top-level discussion about if and how learning outcomes should be reviewed/re-oriented.

Ideally learning outcomes for each week would aim to fulfil the learning outcomes of the whole course. I wrote something in #43 after the recent general meeting, and welcome your comments on it as a basis for our discussion here.

@sofer @NataliaLKB @ZooeyMiller

**Comment by rebeccaradding** [comment]
@eliascodes I don't see your comment in #48...

**Comment by eliascodes** [comment]
@rebeccaradding it's definitely there, Nelson reacted to it, so it can't just be me.

**Comment by rebeccaradding** [comment]
@eliascodes my bad, not sure why I couldn't see it before!

**Comment by eliascodes** [comment]
I want to move this forward because I feel like it will have an impact on other conversations that are being had around the curriculum update process and role of master-reference maintainers (#735).

At the very least @NataliaLKB has seen my comment in #48 and reacted positively. I'm going to use the overall course learning outcomes I outlined in #48 as a basis (but am _very_ happy for other ones to be suggested). Reproducing here for convenience:

> 1. Ability to work effectively in diverse teams
> 2. Ability to adapt to changing circumstances and cope with unanticipated problems
> 3. Ability to together break problems down into smaller sub-problems
> 4. Ability to work across the whole software lifecycle
> 5. Ability to work effectively as a junior full stack developer

IMO, weeks 1-8 would be focussed on outcomes (1), (3), (5), with bits of (2) inevitably. How would each outcome be addressed by the structural elements of the course? Well:

#### 1 
Project teams, research teams, emphasis on pair programming across all materials, workshops designed specifically around pairing etc.

#### 4
20/20/20, mentoring rather than teaching, outlining answers instead of giving them, workshops with hints and prodding rather than hand-holding, linking to official documentation in resources.

#### 5
Subject matter (JS/Node/psql/Express) is relevant to the current local commercial environment

#### 2
Mishaps and general chaos

So in this sense the overall course learning outcomes have a primarily thematic influence on the individual weeks 1-8, with only (5) really determining much about the content.

Thoughts appreciated please @sofer @NataliaLKB @ZooeyMiller @rebeccaradding 🙏

**Comment by mattlub** [comment]
I'm trying to follow this but am confused. Please could someone clarify exactly what is meant by:

"Review scope of learning outcomes for Weeks 1-8" or
"Parameters for learning outcomes for Weeks 1-8"

because it doesn't seem clear to me (as someone who wasn't in the meeting) what is going on/what is the aim here.

**Comment by ZooeyMiller** [comment]
@eliascodes I largely agree with those outcomes for the overall outcomes for the first 8 weeks, and the methods to achieve them (especially a fan of "general chaos"). 

I think with: 

>4
>
>20/20/20, mentoring rather than teaching, outlining answers instead of giving them, workshops with hints and prodding rather than hand-holding, linking to official documentation in resources.

We may need to do more mentor prep in order to achieve this, and this is across all locations. If we consider the way mentors act to be a core part of the method to achieve the desired outcomes of the course, then I think we should ensure that mentors have the materials available to equip themselves with the skills required to do so. 

I think we also need to address the learning outcomes of each week when we have the desired outcomes of the first 8 weeks as a whole in place.

**Comment by eliascodes** [comment]
> We may need to do more mentor prep in order to achieve this

I agree, and on reflection, and discussion with @astroash, many the example outcomes I put in #48 are probably better expressed through the structure of the course (e.g. projects, jigsaw, pairing) and mentoring style (e.g. no hand-holding, 20/20/20) than any specific learning outcomes of each week.

> I think we also need to address the learning outcomes of each week when we have the desired outcomes of the first 8 weeks as a whole in place.

While I agree with this in principle, I think there are clear improvements we can make now with a more informal approach, considering historical feedback and experience of the way the course has been running.

So to try to move this forward again, I've produced a first pass of [target learning outcomes for weeks 1-8](https://trello.com/b/p4ViRX8I). For reference, here are the [current learning outcomes](https://trello.com/b/StQDrB7s).

I arrived at these changes by:
* Trimming the fat:
  * some learning outcomes were just re-stated research questions, 
  * some were effectively duplicates (e.g tdd and red-green-refactor)
  * some were unnecessarily specific (request module / shot / tap-spec)
* explicitly added areas that have been serially neglected and (from memory) come up during several rounds of curriculum feedback (e.g. validation)
* explicitly added some areas that I perceive as weak points of the course (e.g. actually covering callbacks and the async model, debugging)
* Re-ordered a few things (particularly in weeks 2/3)

TLDR; major changes are: (some) ES6 in pre-course, some explicit outcomes based on JS language features, debugging(!), validation, more CSS(!), promises earlier, linting earlier, removal of node-shell stuff.

Note that while I’m suggesting these changes to the learning outcomes, I’m not proposing any time-scale on which changes to the weeks/resources would need to happen (that is, I anticipate some of the changes will be a lot of work and need to involve alumni as well as students and cannot happen over a single curriculum iteration).

I would really like some feedback/criticism of this. Happy to justify individual changes or learning outcomes to those that point them out.

**Comment by jackpandas** [comment]
@eliascodes I think your Trello breakdown works really well. One thing it highlights is how under-developed the learning outcomes for week 7 appear to be, relative to the rest of the course. The scope of week 7 looks about right (I wish I'd done cookies much later than I did), but the sub-units of knowledge in each topic aren't as sharply defined as it is in the other weeks.

As a wider point, it's remarkable quite how much of the curriculum has stayed constant, despite changing entry criteria for FAC, new frameworks, re-ordered weeks etc.

One trans-disciplinary problem with aligned learning is the level at which learning outcomes are defined. I think that is reflected in the FAC curriculum. There's a good (short and free) explanation on defining learning outcomes [here](https://www.heacademy.ac.uk/system/files/resources/id477_aligning_teaching_for_constructing_learning.pdf)

As a long-term idea, it might be worth considering what the FAC curriculum would look like if the basic unit is framed in terms of a problem/question, to which a short answer can be attached, the simplest possible example of a worked answer/design pattern can be attached, and then any number of specific technologies can be attached (to demonstrate how this technology solves this problem, and why).

Let's say, for argument's sake, that each week in the FAC curriculum could be defined in such terms, and that (for argument's sake), the course outcomes require students to learn/understand 50 such questions per week, so roughly 500 in total. "Cookies 101" might be specified as:
- How can I preserve information about a user?
- Why would you want to use cookies?
- What is a cookie?
- Where are cookies located client-side?
- How do you create a cookie?
- How do you set a cookie?
- How do you modify a cookie?
- Why would you add variables to cookies?
- Why are cookies usually set to expire?
- etc etc etc

I can say from experience that this is really, really boring to do, but specifying a course in these terms is perhaps the most helpful thing you can do for lower-performing students to support them.

The benefits for a course like FAC is that a list like this can always be chopped and changed, but most of the questions will remain pretty much the same (maybe a question/problem needs to be broken down into two problems, etc). You could, after one cohort, get feedback to see which questions needed further specification, or which weren't used.

At that point, you can start having course participants/instructors build the knowledge base themselves. The interim solution is to have people identify the resource where they found the answer to the question. The long term solution would be for FAC participants to write up the answer for 1 question per week, with a model solution, to be reviewed by course instructors. This might seem small, but works to FAC's rhythm. If there's 50 questions per module, and 48 students per year in London alone, then you could expect just the London cohorts to answer them all in a year. If there were in fact double, triple, or quadruple the number of questions, then you're still looking at 1-2 years before they are all comprehensively answered. Peer-review, in this model, comes naturally - if something doesn't work, or isn't right, then students will identify it quickly and it can be rectified.

**Comment by eliascodes** [comment]
@jackpandas thanks for your thoughts!

Thanks for linking to the aligned learning primer, it seems to suggest we might need to take a bit more care in designing the end-of-week project briefs.

I'm trying to understand the implications of your question/answer idea, it's not yet clear to me whether it results in a radically different looking course.

Is your suggestion basically to phrase all learning outcomes as questions? I think this is what we've tried to do in the past, but versions on the trello board are just shortened/terse versions. I think we would eventually turn them (back) into questions.

It sounds like you're going further though, when you're talking about building a knowledge base from answers to those questions. Is that knowledge base for use by students, mentors, both? And how would it relate to the existing resources (workshops, morning-challenges) and the master-reference?

**Comment by m4v15** [comment]
Additional to @eliascodes's LO trello boards, I thought I would make a curriculum board of the first 8 weeks, showing all the workshops that happen in each. Can maybe then use this to see what workshop acheives what LO or something and then this can help guide us with the creation of new LO's/curriculum
